pynamodb/connection/base.py

```diff
@@ -1223,49 +1252,20 @@ def query(self,
             operation_kwargs[SELECT] = str(select).upper()
         if scan_index_forward is not None:
             operation_kwargs[SCAN_INDEX_FORWARD] = scan_index_forward
-        tbl = self.get_meta_table(table_name)
-        if tbl is None:
-            raise TableError("No such table: {0}".format(table_name))
-        if index_name:
-            hash_keyname = tbl.get_index_hash_keyname(index_name)
-            if not hash_keyname:
-                raise ValueError("No hash key attribute for index: {0}".format(index_name))
-        else:
-            hash_keyname = tbl.hash_keyname
-        operation_kwargs[KEY_CONDITIONS] = {
-            hash_keyname: {
-                ATTR_VALUE_LIST: [
-                    {
-                        self.get_attribute_type(table_name, hash_keyname): hash_key,
-                    }
-                ],
-                COMPARISON_OPERATOR: EQ
-            },
-        }
-        if key_conditions is not None:
-            for key, condition in key_conditions.items():
-                attr_type = self.get_attribute_type(table_name, key)
-                operator = condition.get(COMPARISON_OPERATOR)
-                if operator not in COMPARISON_OPERATOR_VALUES:
-                    raise ValueError("{0} must be one of {1}".format(COMPARISON_OPERATOR, COMPARISON_OPERATOR_VALUES))
-                operation_kwargs[KEY_CONDITIONS][key] = {
-                    ATTR_VALUE_LIST: [
-                        {
-                            attr_type: self.parse_attribute(value)
-                        } for value in condition.get(ATTR_VALUE_LIST)
-                    ],
-                    COMPARISON_OPERATOR: operator
-                }
+        if name_placeholders:
+            operation_kwargs[EXPRESSION_ATTRIBUTE_NAMES] = self._reverse_dict(name_placeholders)
```

Comment:
would really prefer for this sort of logic to live somewhere in the `expression` module